consensus/src/payload_manager.rs

```diff
@@ -121,82 +133,114 @@ impl PayloadManager {
 
     /// Extract transaction from a given block
     /// Assumes it is never called for the same block concurrently. Otherwise status can be None.
-    pub async fn get_transactions(&self, block: &Block) -> ExecutorResult<Vec<SignedTransaction>> {
+    pub async fn get_transactions(
+        &self,
+        block: &Block,
+    ) -> ExecutorResult<(Vec<SignedTransaction>, usize)> {
```

Comment:
let's have Option<usize> for easier understanding of backwards compatibility and logic, instead of usize::MAX